client/login/magic-login/gravatar/index.tsx

```diff
@@ -0,0 +1,1038 @@
+	useEffect( () => {
+		const eventOptions = { client_id: oauth2ClientId, client_name: oauth2Client?.title };
+
+		if (
+			( prevShowCheckYourEmail.current && ! showCheckYourEmail ) ||
+			( prevShowEmailCodeVerification.current && ! showEmailCodeVerification ) ||
+			( prevShowSecondaryEmailOptions.current && ! showSecondaryEmailOptions )
+		) {
+			recordTracksEvent( 'calypso_gravatar_powered_magic_login_email_form', {
+				...eventOptions,
+				from: currentQueryArguments?.gravatar_from,
+				is_gravatar_flow: isGravatarFlow,
+				is_gravatar_flow_with_email: Boolean(
+					isGravatarFlow && currentQueryArguments?.email_address
+				),
+				is_initial_view: false,
+			} );
+		}
+
+		if ( ! prevShowSecondaryEmailOptions.current && showSecondaryEmailOptions ) {
+			recordTracksEvent(
+				'calypso_gravatar_powered_magic_login_secondary_email_options',
+				eventOptions
+			);
+			recordTracksEvent( 'calypso_gravatar_powered_magic_login_click_main_account', eventOptions );
+		}
+
+		if ( ! prevShowEmailCodeVerification.current && showEmailCodeVerification ) {
+			recordTracksEvent(
+				'calypso_gravatar_powered_magic_login_email_code_verification',
+				eventOptions
+			);
+		}
+
+		if ( ! prevShowCheckYourEmail.current && showCheckYourEmail ) {
+			recordTracksEvent(
+				'calypso_gravatar_powered_magic_login_email_link_verification',
+				eventOptions
+			);
+		}
+
+		// Update refs for next render
+		prevShowCheckYourEmail.current = showCheckYourEmail;
+		prevShowEmailCodeVerification.current = showEmailCodeVerification;
+		prevShowSecondaryEmailOptions.current = showSecondaryEmailOptions;
+	}, [
+		showCheckYourEmail,
+		showEmailCodeVerification,
+		showSecondaryEmailOptions,
+		oauth2ClientId,
+		oauth2Client?.title,
+		isGravatarFlow,
+		currentQueryArguments?.gravatar_from,
+		currentQueryArguments?.email_address,
+	] );
+
+	const resendEmailCountdownId = useRef< NodeJS.Timeout | null >( null );
+	const [ resendEmailCountdown, setResendEmailCountdown ] = useState( RESEND_EMAIL_COUNTDOWN_TIME );
+
+	const resetResendEmailCountdown = () => {
+		if ( ! resendEmailCountdownId.current ) {
+			return;
+		}
+
+		clearInterval( resendEmailCountdownId.current );
+		resendEmailCountdownId.current = null;
+		setResendEmailCountdown( RESEND_EMAIL_COUNTDOWN_TIME );
+	};
+
+	const startResendEmailCountdown = () => {
+		resetResendEmailCountdown();
+
+		resendEmailCountdownId.current = setInterval( () => {
+			setResendEmailCountdown( ( prev ) => {
+				if ( prev <= 1 ) {
+					if ( resendEmailCountdownId.current ) {
+						clearInterval( resendEmailCountdownId.current );
+						resendEmailCountdownId.current = null;
+					}
+					return 0;
+				}
+				return prev - 1;
+			} );
+		}, 1000 );
+	};
+
+	// Ensure interval is cleared on unmount
+	useEffect( () => {
+		return () => {
+			if ( resendEmailCountdownId.current ) {
+				clearInterval( resendEmailCountdownId.current );
+				resendEmailCountdownId.current = null;
+			}
+		};
+	}, [] );
```

Comment:
@wellyshen this part was redone a bit. It should also ensure the interval is cleared when the component unmounts.